test/core/iomgr/timer_list_test.cc

```diff
@@ -151,17 +151,95 @@ void destruction_test(void) {
   GPR_ASSERT(1 == cb_called[2][0]);
 }
 
-int main(int argc, char** argv) {
-  grpc_test_init(argc, argv);
-  grpc_core::ExecCtx::GlobalInit();
+/* Clean up a list with pending timers that simulate long-running-services. */
+/* https://github.com/grpc/grpc/issues/15904 */
+void long_running_service_cleanup_test(void) {
+  grpc_timer timers[4];
   grpc_core::ExecCtx exec_ctx;
-  grpc_determine_iomgr_platform();
-  grpc_iomgr_platform_init();
-  gpr_set_log_verbosity(GPR_LOG_SEVERITY_DEBUG);
-  add_test();
-  destruction_test();
-  grpc_iomgr_platform_shutdown();
+
+  gpr_log(GPR_INFO, "long_running_service_cleanup_test");
+
+  grpc_millis now = grpc_core::ExecCtx::Get()->Now();
+  GPR_ASSERT(now >= 2160000000); /* 25 days in milliseconds */
+  grpc_timer_list_init();
+  grpc_core::testing::grpc_tracer_enable_flag(&grpc_timer_trace);
+  grpc_core::testing::grpc_tracer_enable_flag(&grpc_timer_check_trace);
+  memset(cb_called, 0, sizeof(cb_called));
+
+  grpc_timer_init(
```

Comment:
It would be good to explain in a couple of lines what your test is trying to do..
something like 
```
1) Simulate grpc server start time to 25 days in the past
2) Create 3 timers - one with a deadline 25 days in future, one just 3 milliseconds in future, one way out in the future
3) Simulate 4 milliseconds of elapsed time by changing `now` (cached at step 1) to `now+4`
...
```

Reply:
Done. There is now a more detailed comment at the top of this function.